### ISSUE TITLE
Grid infill shouldn't break the infill lines on intersections

### ISSUE THREAD

**Issue by l29ah** [slic3r/Slic3r#4923]
**Is your feature request related to a problem? Please describe.**
Nowadays Slic3r stops extruding when a Grid infill line crosses another line, resulting in voids that compromise the part rigidity for no good reason.

**Describe the solution you'd like**
Grid infill shouldn't stop extruding when crossing an existing Grid infill line.

**Describe alternatives you've considered**
Making it configurable doesn't seem to make sense since Grid is not used at 100% infill, otherwise there's always space for extra material.

**Comment by lordofhyphens** [comment]
Is this the rectilinear infill mode?

**Comment by l29ah** [comment]
No, as i mentioned five times already, it's the Grid infill mode.

**Comment by l29ah** [comment]
![](https://pstbin.ru/f/25085/)
Arrows where underfills are directly observable, dots where they are present but hidden by other walls.

**Comment by lordofhyphens** [comment]
So here's gcode.ws rendering of a layer of a 20mm cube @ around 3% infill (mostly to make it look easier). 

![infill-1](https://user-images.githubusercontent.com/31754/73322518-8cf20980-420a-11ea-8d89-98c9747011d4.png)
![2](https://user-images.githubusercontent.com/31754/73322517-8cf20980-420a-11ea-872e-08234b959e74.png)
![3](https://user-images.githubusercontent.com/31754/73322515-8cf20980-420a-11ea-824e-987e066bc425.png)
![4](https://user-images.githubusercontent.com/31754/73322513-8cf20980-420a-11ea-8f4f-98f394e420fe.png)
![5](https://user-images.githubusercontent.com/31754/73322512-8cf20980-420a-11ea-8918-0e67ff871f10.png)
![6](https://user-images.githubusercontent.com/31754/73322511-8c597300-420a-11ea-9e6c-e59e75792d28.png)

I think what you are seeing is Slic3r dragging through the existing plastic.

**Comment by lordofhyphens** [comment]
Same cube, 15% infill. Note that Slic3r happily crosses lines. 

![infill15-01](https://user-images.githubusercontent.com/31754/73322894-4cdf5680-420b-11ea-8675-045a3f63c59f.png)
![infill-02](https://user-images.githubusercontent.com/31754/73322897-4e108380-420b-11ea-9a86-703ad0cbab9f.png)
![infill-15-03](https://user-images.githubusercontent.com/31754/73322901-4fda4700-420b-11ea-9ae9-f3e86f342324.png)
![infill-15-04](https://user-images.githubusercontent.com/31754/73322903-510b7400-420b-11ea-9f8c-ee494b9df602.png)
![infill-15-05](https://user-images.githubusercontent.com/31754/73322906-52d53780-420b-11ea-90da-f76512f98654.png)

**Comment by ledvinap** [comment]
This is IMO problem with PET. When crossing perpendicular infil, extrudate gets stripped from nozzle and there is small void just behind.
Alternating infill direction between layers will probably help a lot.

**Comment by bubnikv** [comment]
It happens with any material if you print quickly enough. I suppose it may
have something to do with the non-Newtonian properties of the molten
polymer: The flow is stopped when crossing an existing line, and then it
takes time before the plastic flows with the flow velocity before crossing.
This is a general issue visible not only on the infill, but for example
when filling in the top surfaces with a slight intentional overfill to fill
in the corners between the perimeters and the infill lines, where the flow
is constrained by the already extruded material. One than may see
underextrusion on the following extrusion, if the following extrusion
happens into a free enough space.



st 29. 1. 2020 v 9:30 odesílatel Petr Ledvina <notifications@github.com>
napsal:

**Comment by Trellian** [comment]
> This is IMO problem with PET. When crossing perpendicular infil, extrudate gets stripped from nozzle and there is small void just behind.
> Alternating infill direction between layers will probably help a lot.

I have also experienced this a lot with PET, very little with ABS.
